### ISSUE TITLE
TypeError: req._resetTimeout is not a function

### ISSUE THREAD

**Issue by a178235** [fbbdev/node-fastcgi#4]
I've installed https://github.com/angular/universal and get the following message when a FastCGI connection is made.

/app/node_modules/node-fastcgi/lib/server.js:251
  req._resetTimeout();
      ^

TypeError: req._resetTimeout is not a function
    at Server.roleHandler (/app/node_modules/node-fastcgi/lib/server.js:251:7)
    at Server.recordHandler (/app/node_modules/node-fastcgi/lib/server.js:246:24)
    at null.<anonymous> (/app/node_modules/node-fastcgi/lib/server.js:122:26)
    at emitTwo (events.js:87:13)
    at emit (events.js:172:7)
    at processIncomingData (/app/node_modules/node-fastcgi/node_modules/fastcgi-stream/lib/stream.js:64:10)
    at Socket.<anonymous> (/app/node_modules/node-fastcgi/node_modules/fastcgi-stream/lib/stream.js:82:3)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:146:16)

My index.js is:

var fcgi = require('node-fastcgi');
var server = require('./dist/examples/app/server/server')(__dirname);
module.exports.Server = fcgi.createServer(server).listen('/tmp/node.socket');

**Comment by mnjrupp** [comment]
This might be a versioning issue with NodeJs. What version of Node are you using?

**Comment by a178235** [comment]
Node.js 4.2.1. Your fileServer.js example worked fine.

**Comment by fbbdev** [comment]
The problem is in the express framework. It replaces the prototype of request objects and baffles the FastCGI handler. I'm trying to fix it. They should not do that anyway: they assume everyone is using the http module, which is not necessarily true (as shown by our case).

**Comment by fbbdev** [comment]
Can you please explain me how to reproduce your environment? I don't understand how to install angular/universal and I don't have time to study it.

As a workaround you can run `fcgi.patchHttp()` before importing the angular server. This method patches the `http` module with functions and objects from `node-fastcgi`. It's quite a dirty hack but it should work. Keep in mind that it will make the `http` module unusable for regular http communication (both client and server mode).

Your index.js should become:
```javascript
var fcgi = require('node-fastcgi');
fcgi.patchHttp();
var server = require('./dist/examples/app/server/server')(__dirname);
module.exports.Server = fcgi.createServer(server).listen('/tmp/node.socket');
```

**Comment by fbbdev** [comment]
Please note that this issue is completely resolved in the latest release (v1.1.0)
